api/types/okta.go

```diff
@@ -396,6 +396,79 @@ func (o *OktaAssignmentV1) CheckAndSetDefaults() error {
 	return nil
 }
 
+// IsEqual determines if two okta assignment resources are equivalent to one another.
+func (o *OktaAssignmentV1) IsEqual(i interface{}) bool {
+	ptr, ok := i.(*OktaAssignmentV1)
+	if ok {
+		if o == nil && ptr == nil {
+			return true
+		}
+		if o != nil && ptr != nil {
+			return o.isEqual(*ptr)
+		}
+		return false
+	}
+
+	other, ok := i.(OktaAssignmentV1)
+	if ok {
+		return o.isEqual(other)
```

Comment:
It's a bit strange that we're checking `o` for `nil` above but not here. Same in other similar places.